subxt/src/config/signed_extensions.rs

```diff
@@ -370,6 +371,95 @@ impl<T: Config> SignedExtension<T> for ChargeTransactionPayment {
     type Decoded = Self;
 }
 
+/// The [`SkipCheckIfFeeless`] signed extension.
+#[derive(Debug, DecodeAsType)]
+pub struct SkipCheckIfFeeless<T, S>(S, PhantomData<T>)
+where
+    T: Config,
+    S: SignedExtension<T>;
+
+impl<T, S> SkipCheckIfFeeless<T, S>
+where
+    T: Config,
+    S: SignedExtension<T>,
+{
+    /// The inner signed extension.
+    pub fn inner(&self) -> &S {
+        &self.0
+    }
+}
+
+impl<T, S> ExtrinsicParams<T> for SkipCheckIfFeeless<T, S>
+where
+    T: Config,
+    S: SignedExtension<T>,
+    <S as ExtrinsicParams<T>>::OtherParams: Default,
+{
+    type OtherParams = SkipCheckIfFeelessParams<T, S>;
+    type Error = <S as ExtrinsicParams<T>>::Error;
+
+    fn new<Client: OfflineClientT<T>>(
+        nonce: u64,
+        client: Client,
+        other_params: Self::OtherParams,
+    ) -> Result<Self, Self::Error> {
+        let other_params = match other_params.0 {
+            Some(other_params) => other_params,
+            None => <S as ExtrinsicParams<T>>::OtherParams::default(),
+        };
+
+        let inner_extension = S::new(nonce, client, other_params)?;
+        Ok(SkipCheckIfFeeless(inner_extension, PhantomData))
+    }
+}
+
+impl<T, S> ExtrinsicParamsEncoder for SkipCheckIfFeeless<T, S>
+where
+    T: Config,
+    S: SignedExtension<T>,
+{
+    fn encode_extra_to(&self, v: &mut Vec<u8>) {
+        self.0.encode_extra_to(v);
```

Comment:
Since this can be used with different exts inside, I wonder how easily we could make it use encode_as_type to encode the inner extension based on the metadata. That'd offer extra safety because then we'd get an error if we try to encode an inner signed ext that doesn't match what the node is expecting, whereas without metadata we'd just blindly encode and hope for the best (and prolly get a super obscure decode error back from the node if it's wrong!)

It could enforce that the signed ext impls EncodeAsType too maybe! Not sure :)